exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java

```diff
@@ -33,36 +33,113 @@
 import org.apache.drill.exec.util.BatchPrinter;
 import org.apache.drill.exec.vector.VectorValidator;
 
+import static org.apache.drill.exec.record.RecordBatch.IterOutcome.*;
+
+
 public class IteratorValidatorBatchIterator implements CloseableRecordBatch {
-  static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(IteratorValidatorBatchIterator.class);
+  private static final org.slf4j.Logger logger =
+      org.slf4j.LoggerFactory.getLogger(IteratorValidatorBatchIterator.class);
 
   static final boolean VALIDATE_VECTORS = false;
 
-  private IterOutcome state = IterOutcome.NOT_YET;
+  /** For logging/debuggability only. */
+  private static volatile int instanceCount;
+
+  /** For logging/debuggability only. */
+  private final int instNum;
+  {
+    instNum = ++instanceCount;
```

Comment:
not thread-safe but not a show stopper since it is improving debugging experience.